crates/telemetry/src/lib.rs

```diff
@@ -98,12 +101,33 @@ pub fn init(spin_version: String) -> anyhow::Result<ShutdownGuard> {
 }
 
 fn otel_error_handler(err: opentelemetry::global::Error) {
+    // Track the error count
+    let signal = match err {
+        opentelemetry::global::Error::Metric(_) => "metrics",
+        opentelemetry::global::Error::Trace(_) => "traces",
+        opentelemetry::global::Error::Log(_) => "logs",
+        _ => "unknown",
+    };
+    metrics::monotonic_counter!(spin.otel_error_count = 1, signal = signal);
+
+    // Only log the first error at ERROR level, subsequent errors will be logged at higher levels and rate limited
     static FIRST_OTEL_ERROR: std::sync::Once = std::sync::Once::new();
     FIRST_OTEL_ERROR.call_once(|| {
-        tracing::error!("There has been an error with the OpenTelemetry system, traces and metrics are likely failing to export");
-        tracing::error!("Further OpenTelemetry errors will be logged at DEBUG level")
+        tracing::error!(?err, "OpenTelemetry error");
+        tracing::error!("There has been an error with the OpenTelemetry system. Traces, logs or metrics are likely failing to export. Further OpenTelemetry errors will be logged at WARN level.");
     });
-    tracing::debug!(?err, "OpenTelemetry error");
+
+    // Rate limit the logging of the OTel errors to not occur more frequently on each thread than OTEL_ERROR_INTERVAL
+    const OTEL_ERROR_INTERVAL: Duration = Duration::from_millis(5000);
+    thread_local! {
+        static LAST_OTEL_ERROR: Cell<Instant> = Cell::new(Instant::now() - OTEL_ERROR_INTERVAL);
+    }
+    if LAST_OTEL_ERROR.get().elapsed() > OTEL_ERROR_INTERVAL {
```

Comment:
This is arguably less confusing if you have TRACE enabled (you won't have random WARNs sprinkled around):

```suggestion
    if !tracing::enabled!(tracing::Level::TRACE) && LAST_OTEL_ERROR.get().elapsed() > OTEL_ERROR_INTERVAL {
```